docs/x509.rst

```diff
@@ -718,6 +718,68 @@ X.509 Extensions
 
         :returns: A list of values extracted from the matched general names.
 
+.. class:: CertificatePolicies
+
+    .. versionadded:: 0.9
+
+    The certificate policies extension is a list of one or more
+    :class:`PolicyInformation` instances. The object is iterable to get every
+    instance.
+
+Certificate Policies Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These classes may be present within a :class:`CertificatePolicies` instance.
+
+.. class:: PolicyInformation
+
+    .. versionadded:: 0.9
+
+    Contains a policy identifier and an optional list of qualifiers.
+
+    .. attribute:: policy_identifier
+
+        :type: :class:`ObjectIdentifier`
+
+    .. attribute:: policy_qualifiers
+
+        :type: list
+
+        A list of :class:`PolicyQualifierInfo` objects.
+
+.. class:: PolicyQualifierInfo
+
+    .. versionadded:: 0.9
+
+    .. attribute:: qualifier
+
+        :type: :term:`text` or :class:`UserNotice`
```

Comment:
This is basically represented by this in the RFC:

```
   Qualifier ::= CHOICE {
        cPSuri           CPSuri,
        userNotice       UserNotice }
```

Rather than treating this as a union to a single property we could choose to make this two separate properties where one is not None. That would make usage of this API look like:

```python
if pqi.user_notice:
    # do something
else:
    # do something with pqi.cps_uri
```
instead of
```python
if isinstance(pqi.qualifier, UserNotice):
    # do something
else:
   # do something else
```


Incidentally, this choice problem will appear again with CDP when we get to that.
```
   DistributionPointName ::= CHOICE {
        fullName                [0]     GeneralNames,
        nameRelativeToCRLIssuer [1]     RelativeDistinguishedName }
```